apps/coscrad-frontend/src/components/nav-menu/nav-menu-handler-container.tsx

```diff
@@ -0,0 +1,35 @@
+import { isNull } from '@coscrad/validation-constraints';
+import { Box } from '@mui/material';
+import { useState } from 'react';
+import { NavItemInfo } from './nav-menu-container';
+import { NavMenuPresenter } from './nav-menu-presenter';
+
+interface NavMenuPresenterProps {
+    navItemInfos: NavItemInfo[];
+}
+
+export const NavMenuHandlerContainer = ({ navItemInfos }: NavMenuPresenterProps): JSX.Element => {
+    const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
+
+    const isOpen = !isNull(anchorEl);
+
+    const handleClick = (event: React.MouseEvent<HTMLButtonElement>) => {
+        setAnchorEl(event.currentTarget);
+    };
+
+    const handleClose = () => {
+        setAnchorEl(null);
+    };
+
+    return (
+        <Box>
+            <NavMenuPresenter
+                navItemInfos={navItemInfos}
```

Comment:
The large number of props suggests that these things are too tightly coupled to be separate. In general, I am ok with UX state being present in presenters, but I do like the separation of concerns you were able to create.